version/cobra.go

```diff
@@ -109,10 +109,10 @@ func CobraCommandWithOptions(options CobraOptions) *cobra.Command {
 		},
 	}
 
-	cmd.Flags().BoolVarP(&short, "short", "s", false, "Displays a short form of the version information")
+	cmd.Flags().BoolVarP(&short, "short", "s", false, "Use --short=false to generate JSON")
```

Comment:
`--short=false` is not Json, `-o json` is. I am not sure what to call it though..

Reply:
How about calling the flag json?

Reply:
But it isn't json. It will be something like:
```
version.BuildInfo{Version:"661ca4bcba7cf82208864d3f36a80e1d936712a5-dirty", GitRevision:"661ca4bcba7cf82208864d3f36a80e1d936712a5-dirty", User:"howardjohn", Host:"howardjohn.svl.corp.google.com", GolangVersion:"go1.12.5", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.0-rc.0-109-g661ca4bcb
```

The json output is
```
{
  "clientVersion": {
    "version": "661ca4bcba7cf82208864d3f36a80e1d936712a5-dirty",
    "revision": "661ca4bcba7cf82208864d3f36a80e1d936712a5-dirty",
    "user": "howardjohn",
    "host": "howardjohn.svl.corp.google.com",
    "golang_version": "go1.12.5",
    "hub": "docker.io/istio",
    "status": "Modified",
    "tag": "1.2.0-rc.0-109-g661ca4bcb"
  }
}
```

Reply:
Seems like Go value output. Given that, the command-line message seems apt.

How do we use it? Rather, which output format do we use?

Reply:
The original messages being replaced were created when the defaults were `false`.  Now that the defaults are `true`, users need to know to use `=false` to get the old behavior.

I had forgotten the output was Go source code format!

Instead of documenting the possibility of `--short=false` how about I add a new flag `--json` that actually outputs in JSON and that I deprecate `--short`?  Or we can leave it in, undeprecated.  I don't think anyone appreciates the Golang output and we should consider dumping it.

Reply:
FWIW kubernetes uses the go source format. The go format is about 10x more compact than our json output, because we do pretty printing of the json so each remote component takes 10 lines compared to 1. We already have an `-o json` flag so I am not sure if we should add a `--json` flag as well, or if it is worth the deprecation effort.

If we changed short to output compact json instead of go source that seems fine to me though? I doubt anyone is depending on the format of it